### DISCUSSION TITLE
PaddleOCR部署问题相关

### DISCUSSION THREAD

**Discussion by v587xpt** [hiroi-sora/Umi-OCR#174] [General]
公司也有了类似的项目，要识别身份证。所以接触了百度飞浆-PaddleOCR并经历了一下步骤：（别问我为什么不用大厂的身份证识别接口，问就是：公司不想花money）
1、最开始部署了命令行paddlecor，经理（java开发）要求有api接口，才能方便调用；
2、研究部署了PaddleHub Serving（pip安装），用起来确实不错，但是识别速度很差（公司没有gpu资源，用的cpu），识别一张身份证照片至少十几秒；
3、进一步研究了基于PaddleServing的部署（python pipeline），部署后速度也是由了些许提升，识别一张身份证照片大约5s-10s，但要达到经理的要求还是不够 (╬￣皿￣)；
4、研究、部署基于PaddleServing-C++的部署，走到这一步，始终没能突破；（原因可能有两点：百度的文档太垃圾、我技术太菜）；
后来迫于无奈寻找其他图像识别的项目，发现了Umi-OCR，测试使用后的感受：“真的快”，进而找到了【PaddleOCR-json C++】，算是完成了任务；
对于博主移植linux的想法，我也很乐意参与，但是就怕自己能力有限，毕竟对于linux上 C++的部署还是没能成功，惭愧惭愧。博主如果有用得着的地方，吩咐就行，linux运维工程师；
如果有大佬指点下linux上paddleocr C++部署就更好了；

**Comment by hiroi-sora** [comment]
嗯，我对linux也不是很熟，期待有大佬的参与。

说到你的需求——身份证识别，PPOCR正好有一个分支功能 [PP-Structure（关键信息抽取）](https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/ppstructure/README_ch.md#33-%E5%85%B3%E9%94%AE%E4%BF%A1%E6%81%AF%E6%8A%BD%E5%8F%96) 契合这个方向。贵公司若要在生产环境下使用，则这种垂类模型会比通用OCR模型更高效稳定。

我的PaddleOCR-json支持`PP-Structure`模型库，不过由于目前用不上，代码中屏蔽了相关功能。你可以研究下看看。
